### PR TITLE
Add hoc metric

### PR DESCRIPTION
**Pull request by padjal**
In an attempt to close #188, I have created a new PR for the HOC metric.

### CODE REVIEW

**Comment by yegor256** [comment]
@padjal many thanks for your contribution! Please, see the log above -- `make test` fails.

**Comment by padjal** [comment]
@yegor256 All checks passed. Can you merge this?

File: metrics/hoc.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+base=$(basename "${java}")
+
+# To check that file was added in commit any time
+if git status > /dev/null 2>&1 && test -n "$(git log --oneline -- "${base}")"; then
+    hoc=$(git log -L:"class\s${class_name}:${java}" | grep -E "^[+-].*$" | grep -Ev "^\-\-\-\s\S+$" | grep -Evc "^\+\+\+\s\S+$")
+else
+    hoc=0
+fi
+
+echo "hoc ${hoc} Hits Of Code for ${class_name} class" > "${output}"
```
Comment (yegor256) [inline metrics/hoc.sh:40]: @padjal you don't need to mention the class name here, see: https://github.com/yegor256/cam/tree/master/metrics

File: metrics/hoc.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+set -e
+set -o pipefail
+
+java=$1
+output=$(realpath "$2")
+class_name=$3
```
Comment (yegor256) [inline metrics/hoc.sh:28]: @padjal the third parameter will not be provided to the script, see: https://github.com/yegor256/cam/tree/master/metrics

**Comment by yegor256** [comment]
@padjal perfect, thanks! see some comments above

**Comment by padjal** [comment]
@yegor256 I edited the files according to your comments and fixed the tests. Metric should be ready now.

**Comment by yegor256** [comment]
@padjal we already merged #188

**Pull request closed without merging**